unity/PitayaExample/Assets/Pitaya/Editor/PitayaXcodeProjectUpdater.cs

```diff
@@ -1,58 +1,90 @@
-#if UNITY_IOS || UNITY_EDITOR
+#if UNITY_IOS || UNITY_EDITOR_OSX
+using System.Collections.Generic;
 using UnityEditor;
 using UnityEditor.Callbacks;
 using UnityEditor.iOS.Xcode;
+using System;
 using System.IO;
+using System.Linq;
 
 public class PitayaBuildPostprocessor
 {
-	private const string SIMULATOR_IDENTIFIER = "simulator";
-	private const string DEVICE_IDENTIFIER = "device";
-	private const string FRAMEWORK_ORIGIN_PATH = "Assets/Pitaya/Native/iOS";
-	private const string FRAMEWORK_UPM_ORIGIN_PATH = "Libraries/com.wildlifestudios.nuget.libpitaya/Native/iOS";
-
-	[PostProcessBuild]
-	public static void OnPostprocessBuild(BuildTarget buildTarget, string path)
-	{
-		if (buildTarget == BuildTarget.iOS)
-		{
-			var projPath = path + "/Unity-iPhone.xcodeproj/project.pbxproj";
-
-			PBXProject proj = new PBXProject();
-			proj.ReadFromString(File.ReadAllText(projPath));
-
-			var targetGuid = proj.GetUnityFrameworkTargetGuid();
-
-			// Pitaya should be linked with zlib when on iOS.
-			proj.AddBuildProperty(targetGuid, "OTHER_LDFLAGS", "-lz");
-			
-			// To avoid referencing wrong libraries, the folder containing those
-			// libraries should be hidden depending on which target the build
-			// is running (device or simulator).
-			foreach (var file in Directory.GetFiles(GetPathToHide(path))) {
-				var fileGuid = proj.FindFileGuidByProjectPath(file.Substring(path.Length + 1));
-				// Files will be kept in the build folder, but won't be referenced in the built project
-				proj.RemoveFile(fileGuid);
-			}
-
-			File.WriteAllText(projPath, proj.WriteToString());
-		}
-	}
-
-	private static string GetPathToHide(string projectPath)
-	{
-		var originPath = Path.Combine(projectPath, FRAMEWORK_ORIGIN_PATH);
-		if (!Directory.Exists(originPath))
-		{
-			originPath = Path.Combine(projectPath, FRAMEWORK_UPM_ORIGIN_PATH);
-		}
-
-		if(PlayerSettings.iOS.sdkVersion == iOSSdkVersion.DeviceSDK)
-		{
-			return Path.Combine(originPath, SIMULATOR_IDENTIFIER);
-		}
-
-		return Path.Combine(originPath, DEVICE_IDENTIFIER);
-	}
+    private const string SIMULATOR_IDENTIFIER = "simulator";
+    private const string DEVICE_IDENTIFIER = "device";
+    private const string PITAYA_IOS_PATH = "PitayaNativeLibraries/iOS";
+
+    [PostProcessBuild]
+    public static void OnPostprocessBuild(BuildTarget buildTarget, string path)
+    {
+        if (buildTarget == BuildTarget.iOS)
+        {
+            var projPath = path + "/Unity-iPhone.xcodeproj/project.pbxproj";
+            var currPath = Directory.GetCurrentDirectory();
+            var lookupPaths = new string[]
+            {
+                Path.Combine(currPath, "Assets"), // relative to project path
+                Path.Combine(currPath, "Library", "PackageCache"), // relative to project path
+                Path.Combine(path, "Libraries") // relative to build path
+            };
```

Comment:
nit: let's move this section closed to where it's used (around line 38), specially after the PBXProject instance is created.